src/parser/parseSVGDocument.ts

```diff
@@ -14,39 +13,45 @@ import { getGradientDefs } from './getGradientDefs';
 import { hasAncestorWithNodeName } from './hasAncestorWithNodeName';
 import { parseElements } from './parseElements';
 import { parseUseDirectives } from './parseUseDirectives';
-
+import type { TSvgParsedCallback, TSvgReviverCallback } from './typedefs';
+import type { LoadImageOptions } from '../util/misc/objectEnlive';
 /**
  * Parses an SVG document, converts it to an array of corresponding fabric.* instances and passes them to a callback
  * @static
  * @function
  * @memberOf fabric
- * @param {SVGDocument} doc SVG document to parse
- * @param {Function} callback Callback to call when parsing is finished;
- * It's being passed an array of elements (parsed from a document).
- * @param {Function} [reviver] Method for further parsing of SVG elements, called after each fabric object created.
- * @param {Object} [parsingOptions] options for parsing document
- * @param {String} [parsingOptions.crossOrigin] crossOrigin settings
- * @param {AbortSignal} [parsingOptions.signal] see https://developer.mozilla.org/en-US/docs/Web/API/AbortController/signal
+ * @param {HTMLElement} doc SVG document to parse
+ * @param {TSvgParsedCallback} callback Invoked when the parsing is done, with null if parsing wasn't possible of with the list of parsed objects.
+ * TSvgParsedCallback receive also `allElements` array as last argument. This is the full array from the list of svg nodes available in the document.
+ * You may want to use it if you are trying to regroup the objects as they were originally grouped in the SVG. ( This was the reason why it was added )
+ * @param {TSvgReviverCallback} [reviver] Extra callback for further parsing of SVG elements, called after each fabric object created.
+ * Takes as input the original svg element and the fabricObject generated as arguments. Used to inspect extra properties eventually not parsed by fabricJS
+ * @param {Object} [options] Object containing options for parsing
+ * @param {String} [options.crossOrigin] crossOrigin crossOrigin setting to use for external resources
+ * @param {AbortSignal} [options.signal] handle aborting, see https://developer.mozilla.org/en-US/docs/Web/API/AbortController/signal
```

Comment:
not necessary
- it isn't recognized (at least not by VSCODE)
- it is documented by `LoadImageOptions`

Reply:
Until i know how the documentation is generated and what does it need, i ll keep extensive maybe redundant comments.
When the tsdocs generation works we will better know what is seen and what not.